test/e2e/cert_rotation.go

```diff
@@ -72,40 +58,38 @@ func certRotation(ctx context.Context, inputGetter func() CertRotationInput) {
 	}
 	for _, secretName := range secretList {
 		err := clientSet.CoreV1().Secrets(ironicNamespace).Delete(ctx, secretName, metav1.DeleteOptions{})
-		Expect(err).ToNot(HaveOccurred(), "Cannot detele this secret: %s", secretName)
+		Expect(err).ToNot(HaveOccurred(), "cannot delete this secret: %s", secretName)
 	}
-
-	By("Wait for containers in the ironic pod to be restarted")
+	By("Check if the secrets are deleted")
+	for _, secretName := range secretList {
+		_, err := clientSet.CoreV1().Secrets(ironicNamespace).Get(ctx, secretName, metav1.GetOptions{})
+		Expect(err).To(HaveOccurred(), secretName)
+		Expect(apierrors.IsNotFound(err)).To(BeTrue())
+	}
+	By("Wait until secrets are recreated")
 	Eventually(func() error {
-		ironicPod, err := getPodFromDeployment(ctx, clientSet, ironicDeployment, ironicNamespace)
-		if err != nil {
-			return err
-		}
-		// check for container in containerNumRestart list
-		for container := range containerNumRestart {
-			notFound := true
-			for _, ironicContainer := range ironicPod.Status.ContainerStatuses {
-				if ironicContainer.Name == container {
-					notFound = false
-					break
-				}
-			}
-			if notFound {
-				return fmt.Errorf("%s container does not exist in Ironic pod", container)
-			}
+		ironicSecretCreated := isSecretRecreated(ctx, clientSet, ironicNamespace, secretList[0])
```

Comment:
I don't think function is needed here, it's just one line.  
```suggestion
	_, err := clientset.CoreV1().Secrets(namespace).Get(ctx, secretList[0], metav1.GetOptions{})
	Expect(err).ToNot(HaveOccurred(), "secret is being recreated")
```

Reply:
done